src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1511,26 +1527,20 @@
     Ex: 
      (a:=1, b:= (c:=1, d:=Nothing))  does not have a natural type, because "Nothing" does not have one
     -->
-    <Field Name="Type" Type="TypeSymbol?" Override="true"/>
-    <Field Name="ArgumentNamesOpt" Type="ImmutableArray&lt;string&gt;" Null="allow"/>
-
-    <!-- Which argument names were inferred (as opposed to explicitly provided)? -->
-    <Field Name="InferredNamesOpt" Type="ImmutableArray&lt;bool&gt;" Null="allow"/>
+    <Field Name="Type" Type="TypeSymbol" Override="true" Null="allow"/>
   </Node>
 
   <!-- 
   Tuple literals can convert to the target type.
-  Once converted to a target type, they cannot be target-typed again. 
-  The Converted tuple literal is one which has been already converted to a target type.
+  Once converted to a target type, they cannot be target-typed again.
+  The Converted tuple literal is one which has already been converted to a target type.
   Converted tuple literal always has a type.
   -->
   <Node Name="BoundConvertedTupleLiteral" Base="BoundTupleExpression">
-    <!-- Converted tuple must have a type -->
-    <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
```

Comment:
Why remove this? Seems good to enforce that converted tuple literal must have a type.

Reply:
A converted tuple literal is not required to have a type in some scenarios in which they are used to represent tuple equality, for example when there is a `null` on both sides like `(1, null) == (1, null)`.  Why we allow that I do not know, but we do allow it.

---
In reply to: [301718699](https://github.com/dotnet/roslyn/pull/37052#discussion_r301718699) [](ancestors = 301718699)